### PR TITLE
Fix Upgrade Downgrade manual backup dependencies install

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This Pull Request fixes the installation of dependencies in the manual backup upgrade downgrade test. It uses MySQL8.0.29 instead of 8.0.28. The issue can be observed in the following build:

- https://github.com/vitessio/vitess/runs/7980238746?check_suite_focus=true

This PR unblocks the following PRs:
- https://github.com/vitessio/vitess/pull/11052
- https://github.com/vitessio/vitess/pull/11078
- https://github.com/vitessio/vitess/pull/11077
- https://github.com/vitessio/vitess/pull/11068
- https://github.com/vitessio/vitess/pull/11051
- https://github.com/vitessio/vitess/pull/11040
- https://github.com/vitessio/vitess/pull/10928

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by frouioui** [comment]
No backports needed to `release-12.0` as they were no upgrade/downgrade tests at that time.

File: .github/workflows/upgrade_downgrade_test_backups_manual.yml [hunk 1]
```diff
@@ -126,33 +126,36 @@ jobs:
         sudo deluser mysql
         sudo rm -rf /var/lib/mysql
         sudo rm -rf /etc/mysql
-
         # Install MySQL 8.0
         ####
-        ## Pin the MySQL version at 8.0.29 as Vitess 13.0 does not have the fix to support
-        ## backups of 8.0.30+ and no backport to v13.0 is currently planned.
-        ##   See: https://github.com/vitessio/vitess/pull/10847
+        ## Temporarily pin the MySQL version at 8.0.29 as Vitess 14.0.1 does not have the fix to support
```
Comment (mattlord) [inline .github/workflows/upgrade_downgrade_test_backups_manual.yml:131]: To be pedantic we should change the comment to reflect reality too. Will approve to unblock you.

**Review by mattlord — APPROVED** [review]

**Comment by GuptaManan100** [comment]
I already have a commit fixing this in release-13.0 https://github.com/vitessio/vitess/pull/11040, so backport shouldn't be required as long as we merge that PR.

**Comment by frouioui** [comment]
> I already have a commit fixing this in release-13.0 #11040, so backport shouldn't be required as long as we merge that PR.

Awesome, thank you @GuptaManan100.

**Pull request merged**